extensions/.import-restrictions

```diff
@@ -9,6 +9,7 @@ rules:
   - github.com/gardener/gardener/pkg/client/kubernetes
   - github.com/gardener/gardener/pkg/controllerutils
   - github.com/gardener/gardener/pkg/extensions
+  - github.com/gardener/gardener/pkg/client/core/clientset/versioned
```

Comment:
Why is this needed now?

Reply:
For migration e2e test, we are creating a binding explicitly

Reply:
Fair enough, but how is the migration e2e test related to this `extensions` directory? I don't see where any package in `extensions/...` imports `github.com/gardener/gardener/pkg/client/core/clientset/versioned`.

Reply:
Quoting your msg in slack when this error first occured,
"The reason seems to be that
https://github.com/gardener/gardener/blob/master/extensions/test/testmachinery/healthcheck/healthcheck_operation.go#L30 imports "[github.com/gardener/gardener/test/framework](http://github.com/gardener/gardener/test/framework)"
In test/framework/gardener_utils.go you now import "[github.com/gardener/gardener/pkg/client/core/clientset/versioned](http://github.com/gardener/gardener/pkg/client/core/clientset/versioned)"
This is not allowed in https://github.com/gardener/gardener/blob/master/extensions/.import-restrictions"

Reply:
I think we should move merge the `extensions/test` directory into the `test` directory on root level:

- https://github.com/gardener/gardener/tree/master/extensions/test/testmachinery goes into https://github.com/gardener/gardener/tree/master/test/testmachinery/extensions
- https://github.com/gardener/gardener/tree/master/extensions/test/integration goes into https://github.com/gardener/gardener/tree/master/test/integration/extensions/controller

Then we can drop this import restriction again.

Reply:
Does it make sense to do this in a seperate PR?

Reply:
Done with #6158

Reply:
#6158 is approved and will be merged soon. Probably you can then rebase this PR on top of `master` again.